.github/workflows/pr.yaml

```diff
@@ -291,6 +294,64 @@ jobs:
     steps:
       - run: echo "All done"
 
+  enterprise-pr:
+    name: Make enterprise PR
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+    needs:
+      - ci-publish-js-lib
+      - finish-ci-merge
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.18
+      - name: Configure git for private modules
+        env:
+          GITHUB_BUILD_USERNAME: wge-build-bot
+          GITHUB_BUILD_TOKEN: ${{ secrets.WGE_BUILD_BOT_PR_TOKEN }}
+        run: git config --global url."https://${GITHUB_BUILD_USERNAME}:${GITHUB_BUILD_TOKEN}@github.com".insteadOf "https://github.com"
+      - name: Check out enterprise
+        uses: actions/checkout@v3
+        with:
+          repository: 'weaveworks/weave-gitops-enterprise'
+          token: ${{ secrets.WKS_CI_TEST_BOT_PR_TOKEN }}
+          ref: main
+          fetch-depth: 0
+      - name: Fetch unmerged changes
+        run: |
+          git config user.name wks-ci-test-bot
+          git config user.email wks-ci-test-bot@weave.works
+          git fetch origin
+          if git rev-parse -q --verify origin/track-latest-oss; then
+              git merge origin/track-latest-oss
+          fi
+      - name: Upgrade main version
+        run: |
+          go get -u github.com/weaveworks/weave-gitops@${{ github.event.after }}
+          go mod tidy
+          cd ui-cra && yarn add @weaveworks/weave-gitops@npm:@weaveworks/weave-gitops-main@${{ needs.ci-publish-js-lib.outputs.js-version }}
```

Comment:
🔥 !

Wicked :)

Great start, we might wanna do a `yarn test -u` to update snapshots and things too. (follow up)